tests/Moq.Tests/ProtectedAsMockFixture.cs

```diff
@@ -213,6 +213,89 @@ public void SetupSequence_can_setup_actions()
 			Assert.IsType<InvalidOperationException>(exception);
 		}
 
+		[Fact]
+		public void SetUpSet_should_setup_setters()
+		{
+			this.protectedMock.SetupSet(fish => fish.ReadWritePropertyImpl = 999).Throws(ExpectedException.Instance);
+
+			mock.Object.ReadWriteProperty = 123;
+
+			Assert.Throws<ExpectedException>(() => mock.Object.ReadWriteProperty = 999);
+		}
+
+		[Fact]
+		public void SetUpSet_should_setup_setters_with_property_type()
+		{
+			int value = 0;
+			this.protectedMock.SetupSet<int>(fish => fish.ReadWritePropertyImpl = 999).Callback(i => value = i);
+
+			mock.Object.ReadWriteProperty = 123;
+			Assert.Equal(0, value);
+
+			mock.Object.ReadWriteProperty = 999;
+			Assert.Equal(999, value);
+		}
+
+		[Fact]
+		public void SetUpSet_should_work_recursively()
+		{
+			this.protectedMock.SetupSet(f => f.Nested.Value = 999).Throws(ExpectedException.Instance);
+
+			mock.Object.GetNested().Value = 1;
+			
+			Assert.Throws<ExpectedException>(() => mock.Object.GetNested().Value = 999);
+		}
+
+		[Fact]
+		public void SetUpSet_Should_Work_With_Indexers()
+		{
+			this.protectedMock.SetupSet(
+				o => o[
+					It.IsInRange(0, 5, Range.Inclusive),
+					It.IsIn("Bad", "JustAsBad")
+				] = It.Is<int>(i => i > 10)
+			).Throws(ExpectedException.Instance);
+
+			mock.Object.SetMultipleIndexer(1, "Ok", 999);
+
+			Assert.Throws<ExpectedException>(() => mock.Object.SetMultipleIndexer(1, "Bad", 999));
+
+		}
+
+		[Fact]
+		public void VerifySet_Should_Work()
+		{
+			void VerifySet(Times? times = null,string failMessage = null)
+			{
+				this.protectedMock.VerifySet(
+				o => o[
+					It.IsInRange(0, 5, Moq.Range.Inclusive),
+					It.IsIn("Bad", "JustAsBad")
+				] = It.Is<int>(i => i > 10),
+				times,
+				failMessage
+				);
+			}
+			VerifySet(Times.Never());
+
+			mock.Object.SetMultipleIndexer(1, "Ok", 1);
+			VerifySet(Times.Never());
+
+			Assert.Throws<MockException>(() => VerifySet()); // AtLeastOnce
+
+			mock.Object.SetMultipleIndexer(1, "Bad", 999);
+			VerifySet(); // AtLeastOnce
+			
+			mock.Object.SetMultipleIndexer(1, "JustAsBad", 12);
+			VerifySet(Times.Exactly(2));
+
+			Assert.Throws<MockException>(() => VerifySet(Times.AtMostOnce()));
+
+			var mockException = Assert.Throws<MockException>(() => VerifySet(Times.AtMostOnce(),"custom fail message"));
+			Assert.StartsWith("custom fail message", mockException.Message);
+
```

Comment:
Detail: for consistency with existing formatting, please revert this superfluous blank line.